### PR TITLE
[Feature](bangc-ops): rm mlu_op_kerner.h and modify some ops code

### PR DESCRIPTION
**Pull request by wushaoqiang2168**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

remove mlu_op_kerner.h and modify some ops code.

## 2. Modification

- delete mlu_op_kernel.h
- 涉及算子：abs;active_rotated_filter;ball_query;box_iou_rotated;carafe;deform_roi_pool;div;fill_zero;generate_proposals_v2;get_indice_pairs;log;moe_dispatch_backward_data;moe_dispatch_backward_gate;moe_dispatch_forward;nms_rotated;points_in_boxes;poly_nms;prior_box;psamask;psroipool;roi_align_rotated;roi_crop;roiaware_pool3d;roipoint_pool3d;rotated_feature_align;sqrt;
tensor_stride_process;three_interpolate;three_nn_forward;voxel_pooling_forward;voxelization;yolo_box
- 其他修改：涉及 mlu_op_kernel.h 相关内容，均已删除

## 3. Test Report
- 测试脚本：test_op.sh
`./mluop_gtest --gtest_filter="abs*:active_rotated_filter*:ball_query*:box_iou_rotated*:carafe*:deform_roi_pool*:div*:fill_zero*:generate_proposals_v2*:get_indice_pairs*:log*:moe_dispatch_backward_data*:moe_dispatch_backward_gate*:moe_dispatch_forward*:nms_rotated*:points_in_boxes*:poly_nms*:prior_box*:psamask*:psroipool*:roi_align_rotated*:roi_crop*:roiaware_pool3d*:roipoint_pool3d*:rotated_feature_align*:sqrt*:tensor_stride_process*:three_interpolate*:three_nn_forward*:voxel_pooling_forward*:voxelization*:yolo_box*" --cases_dir=$1`

- MLU370X4
>  test_op.sh /data/mlu-ops/release_test/【通过】
>  test_op.sh /data/mlu-ops/release_temp/ 【通过】

- MLU370M8

>  test_op.sh /data/mlu-ops/release_test/【通过】
>  test_op.sh /data/mlu-ops/release_temp/ 【通过】

- MLU590H8

> test_op.sh /data/mlu-ops/release_test/ 【通过】
> test_op.sh /data/mlu-ops/release_temp/【通过】

- MLU590M9

> test_op.sh /data/mlu-ops/release_test/【通过】
> test_op.sh /data/mlu-ops/release_temp/【通过】

### CODE REVIEW

**Review by defei-coder — APPROVED** [review]

**Review by mahxn0 — APPROVED** [review]

**Comment by mahxn0** [comment]
prior_box,generator_proposals_v2,box_overlaps,voxelization　review完毕．

**Comment by starkhu** [comment]
moe-dispatch-forward没问题

**Review by starkhu — APPROVED** [review]

**Review by DanieeelLiu — APPROVED** [review]
active_rotated_filter_forward, roipoint_pool3d, three_nn_forward, psamask_forward, psamask_backward review 完毕。

**Review by caixianzhang — APPROVED** [review]

**Review by tudejiang79 — APPROVED** [review]

**Comment by tudejiang79** [comment]
psroipool、roi_crop、roi_align_rotated、rotated_feature_align、fill_zero没有问题

**Review by WangYuan-cn — APPROVED** [review]

**Review by liuyuan1-v — APPROVED** [review]

**Comment by liuyuan1-v** [comment]
nms_rotated 没问题

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 1]
```diff
@@ -165,11 +154,11 @@ __mlu_global__ void MLUKernelThreeInterpolateForward(
   /*
    * NRAM partition
    *  |-----------------------------------------------------------------------------------|
-   *  |           nram_features                  |        nram_features_transpose         |
```
Comment (ArtIntAI) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:168]: 这块的注释当前用cpp lint会打乱这个大小排布，建议恢复成原样，不用工具lint

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 2]
```diff
@@ -385,11 +371,14 @@ __mlu_global__ void MLUKernelThreeInterpolateBackward(
   /*
    * NRAM partition
    *  |-------------------------------------------------------------------------------------------|
-   *  |           nram_grad_output                  |        nram_grad_output_transpose           |
+   *  |           nram_grad_output                  | nram_grad_output_transpose
```
Comment (ArtIntAI) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:374]: 同上，建议不用工具lint这块的注释。

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 3]
```diff
@@ -553,42 +538,36 @@ __mlu_global__ void MLUKernelThreeInterpolateBackward(
...
-  MLUKernelThreeInterpolateBackward<<<k_dim, k_type, queue>>>(
-      (float *)grad_output, (int *)indices, (float *)weights, b, c, m, n,
-      c_limit_size, m_limit_size, n_limit_size, (float *)grad_features);
+    mluOpDataType_t d_type, const void *features, const void *indices,
+    const void *weights, const int b, const int c, const int m, const int n,
+    const int c_limit_size, const int m_limit_size, const int n_limit_size,
+    void *output) {
+  if (d_type == MLUOP_DTYPE_HALF) {
+    MLUKernelThreeInterpolateForward<<<k_dim, k_type, queue>>>(
+        (half *)features, (int *)indices, (half *)weights, b, c, m, n,
```
Comment (ArtIntAI) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:549]: 这里感觉每个函数都没有统一：
这里是判断HALF,其他进float分支
但是abs算子是判断FLOAT，其他进half分支
也有两个都判断的，其他提醒Not implemented（感觉这种更完备一点，统一成这种？）

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 1 above)
Comment (wushaoqiang2168) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu]: 已修改会原来的

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 2 above)
Comment (wushaoqiang2168) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu]: 已改回

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 3 above)
Comment (wushaoqiang2168) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:546]: 这里都是按照原先host侧来改的

**Review by ArtIntAI — APPROVED** [review]

**Comment by YunyiGUANG** [comment]
carafe review完毕

**Review by yangdian0315 — APPROVED** [review]

**Pull request merged**